### ISSUE TITLE
Multi-row tabbed panel

### ISSUE THREAD

**Issue by DavidA2014** [kivy/kivy#3994]
I want to show a TabbedPanel with 6 tabs arranged as 2 rows x 3 columns. The code below achieves that in Kivy 1.9.1-dev.  But in 1.9.1 release and 1.9.2-dev this code just shows a single row containing all 6 tabs.  I think this is a bug - the result should be 2 rows x 3 columns.

Tested with Python 3.4 and 1.9.2-dev on Fedora 22.

```
from kivy.config import Config
Config.set('kivy', 'log_level', 'debug')

from kivy.app import App
from kivy.lang import Builder
from kivy.factory import Factory
from kivy.uix.boxlayout import BoxLayout

kv = """
<NewB@Button>:
    text: 'New button'

<TabbedPanelStrip>:
    rows: 2
    cols: 3

<Test>:
    orientation: 'vertical'

    TabbedPanel:
        id: t1
        do_default_tab: False

        TabbedPanelItem:
            id: i1
            text: 'i1'
        TabbedPanelItem:
            id: i2
            text: 'i2'
        TabbedPanelItem:
            id: i3
            text: 'i3'
        TabbedPanelItem:
            id: i4
            text: 'i4'
        TabbedPanelItem:
            id: i5
            text: 'i5'
        TabbedPanelItem:
            id: i6
            text: 'i6'
"""

Builder.load_string(kv)


class Test(BoxLayout):
    def __init__(self, *args, **kwargs):
        super(Test, self).__init__(*args, **kwargs)
        self.callback()

    def callback(self):
        self.add_widget(Factory.NewB())


class TestApp(App):
    def build(self):
        return Test()

if __name__ == '__main__':
    TestApp().run()
```

**Comment by dessant** [comment]
1.8.0 also shows 2 rows in the header, while master isn't.

**Comment by tshirtman** [comment]
git bisect says it's commit 43680041da7ddbac0458bcc04ac33653feb3bfd2

**Comment by dessant** [comment]
@matham, this is also a bug affecting gridlayout, in case you are interested. :yum:

**Comment by d-m-bailey** [comment]
There may be a partial work around.

```

<TabbedPanelStrip>:
    rows: app.tab_rows
    cols: app.tab_cols

class TestApp(App):
    tab_rows = NumericProperty(2)
    tab_cols = NumericProperty(3)

```

Be careful how you set the tab_width and tab_height for the TabbedPanel. See #4339

**Comment by matham** [comment]
The real question is WTF: https://github.com/kivy/kivy/blob/master/kivy/uix/tabbedpanel.py#L467 ?

**Comment by matham** [comment]
Also, ideally the user should not provide styling to the `TabbedPanelStrip` class which is then instantiated because that affect `TabbedPanelStrip` globally. Rather, `TabbedPanel` should have a `panel_strip_cls` that is used to instantiate the strip. That way the user can provide styling. Even better if it allows passing a dict of kwargs.

**Comment by d-m-bailey** [comment]
How about adding `tab_rows, tab_cols` (default 1, 99) property to `TabbedPanel` and use those properties to instantiate `TabbedPanelStrip`?  Or does that only solve part of the problem?

**Comment by matham** [comment]
That solves this problem, but it is a band-aid because it doesn't solve the fundamental problem that the panel strip class cannot be set by the user. It also introduces duplicate variables for something that already exists (`tab_rows`/`rows`). The proper solution is to add a new variable for the strip class with the base class styled to rows being 1, which can be overwritten.

**Comment by akshayaurora** [comment]
I agree  with matham here, the right solution is to start using a strip_cls
which the user can override.
